src/JustSaying/JustSayingBus.cs

```diff
@@ -289,4 +334,93 @@ public void Dispose()
         _startLock?.Dispose();
         _loggerFactory?.Dispose();
     }
+
+    public Task PublishAsync(IEnumerable<Message> messages, CancellationToken cancellationToken)
+        => PublishAsync(messages, null, cancellationToken);
+
+    public Task PublishAsync(IEnumerable<Message> messages, PublishBatchMetadata metadata, CancellationToken cancellationToken)
+    {
+        if (!_busStarted && _startupTasks.Count > 0)
+        {
+            throw new InvalidOperationException("There are pending startup tasks that must be executed by calling StartAsync before messages may be published.");
+        }
+
+        var tasks = new List<Task>();
+        foreach (IGrouping<Type, Message> group in messages.GroupBy(x => x.GetType()))
+        {
+            IMessageBatchPublisher publisher = GetBatchPublishersForMessageType(group.Key);
+            tasks.Add(PublishAsync(publisher, group.ToList(), metadata, 0, group.Key, cancellationToken));
+        }
+
+        return Task.WhenAll(tasks);
```

Comment:
Is there an overload of this that supports passing the `CancellationToken` through?